File: hh-back/src/main/java/gloit/hiperionida/helios/model/AdelantoModel.java
package gloit.hiperionida.helios.model;

import gloit.hiperionida.helios.util.model.AbsAuditoriaModel;

import lombok.*;

import jakarta.persistence.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = false)
@Entity
@Getter
@Setter
@SuperBuilder
@Table(name = "adelanto")
public class AdelantoModel extends AbsAuditoriaModel {
    private Long id;
    private LocalDateTime fecha;
    private Double monto;
    @Column(columnDefinition = "TEXT")
    private String notas;
    private Boolean rendido;

    private Long cajaId;
    private Long conductorId;
    private Long reciboId;

    public AdelantoModel() {
    }

    public AdelantoModel(Long id, Long creadorId, LocalDateTime creada, Long modificadorId, LocalDateTime modificada, Long eliminadorId, LocalDateTime eliminada, Long id1, LocalDateTime fecha, Double monto, String notas, Boolean rendido, Long cajaId, Long conductorId, Long reciboId) {
        super(id, creadorId, creada, modificadorId, modificada, eliminadorId, eliminada);
        this.id = id1;
        this.fecha = fecha;
        this.monto = monto;
        this.notas = notas;
        this.rendido = rendido;
        this.cajaId = cajaId;
        this.conductorId = conductorId;
        this.reciboId = reciboId;
    }
}

File: hh-front/src/models/adelanto_model.js
import { UsuarioModel } from './usuario_model'
import { CajaModel } from './caja_model'

export class AdelantoModel {
  constructor () {
    this.id = ''
    this.descripcion = ''
    this.fecha = ''
    this.monto = ''
    this.notas = ''
    this.recibo = ''
    this.rendido = ''
    this.caja = new CajaModel()

    this.creador = new UsuarioModel()
    this.creada = ''
    this.modificador = new UsuarioModel()
    this.modificada = ''
    this.eliminador = new UsuarioModel()
    this.elimianda = ''
  }
}
